processing/src/main/java/io/druid/segment/data/CachingIndexed.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.metamx.common.Pair;
+import com.metamx.common.logger.Logger;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class CachingIndexed<T> implements Indexed<T>, Closeable
+{
+  public static final int INITIAL_CACHE_CAPACITY = 16384;
+
+  private static final Logger log = new Logger(CachingIndexed.class);
+
+  private final GenericIndexed<T>.BufferIndexed indexed;
+  private final SizedLRUMap<Integer, T> cachedValues;
+  
+  public CachingIndexed(GenericIndexed<T>.BufferIndexed indexed, final int lookupCacheSize)
+  {
+    this.indexed = indexed;
+
+    if(lookupCacheSize > 0) {
+      log.debug("Allocating column cache of max size[%d]", lookupCacheSize);
+      cachedValues = new SizedLRUMap<>(INITIAL_CACHE_CAPACITY, lookupCacheSize);
```

Comment:
`SizedLRUMap` does not look thread safe. Upon cursory view of the construction below, it would seem like the thread safety were dependent on the dictionary, which is not the case. Can you either make this class thread safe, or mark it as NOT thread safe?